From 93482f8ad273473317fd62973b4d18ade99e9bea Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:28 -0400
Subject: [ethernet] qlcnic: Fix to send INIT_NIC_FUNC as first mailbox

Message-id: <1408626422-3941-105-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88628
O-Subject: [PATCH RHEL 7.1 104/138] qlcnic: Fix to send INIT_NIC_FUNC as first mailbox.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 4d52e1e8d1e198962dcbfabf9c06425c38eb23d0

o INIT_NIC_FUNC should be first mailbox sent. Sending DCB capability and
  parameter query commands after that command.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index d211af7..3ca3118 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -2355,7 +2355,9 @@ int qlcnic_83xx_init(struct qlcnic_adapter *adapter, int pci_using_dac)
   goto disable_mbx_intr;
 
  qlcnic_83xx_clear_function_resources(adapter);
+ qlcnic_dcb_enable(adapter->dcb);
  qlcnic_83xx_initialize_nic(adapter, 1);
+ qlcnic_dcb_get_info(adapter->dcb);
 
  /* Configure default, SR-IOV or Virtual NIC mode of operation */
  err = qlcnic_83xx_configure_opmode(adapter);
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_dcb.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_dcb.c
index 7d4f549..a51fe18 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_dcb.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_dcb.c
@@ -330,8 +330,6 @@ static int __qlcnic_dcb_attach(struct qlcnic_dcb *dcb)
   goto out_free_cfg;
  }
 
- qlcnic_dcb_get_info(dcb);
-
  return 0;
 out_free_cfg:
  kfree(dcb->cfg);
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 17236f4..87edac6 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2525,8 +2525,6 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
   goto err_out_free_hw;
  }
 
- qlcnic_dcb_enable(adapter->dcb);
-
  if (qlcnic_read_mac_addr(adapter))
   dev_warn(&pdev->dev, "failed to read mac addr\n");
 
@@ -2546,7 +2544,10 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
     "Device does not support MSI interrupts\n");
 
  if (qlcnic_82xx_check(adapter)) {
+  qlcnic_dcb_enable(adapter->dcb);
+  qlcnic_dcb_get_info(adapter->dcb);
   err = qlcnic_setup_intr(adapter);
+
   if (err) {
    dev_err(&pdev->dev, "Failed to setup interrupt\n");
    goto err_out_disable_msi;
-- 
1.7.1